File: crc.js
var fs = require('fs');
var f = process.argv[2];

binary = fs.readFileSync(f);

function toHexString(byteArray) {
  return Array.from(byteArray, function(byte) {
    return ('0' + (byte & 0xFF).toString(16)).slice(-2);
  }).join('')
}

const crc16 = (bytes) => {
  let result = 65258;
  
  for (const b of bytes) {
    console.log(b)
    return ""
    let b2 = (((result & 0xff) << 8) | ((0xFF00 & result) >> 8)) ^ (b & 0xff);
    let b3 = b2 ^ ((b2 & 0xff) >> 4);
    let b4 = b3 ^ (((b3 & 0xff) << 8) << 4);
    result = b4 ^ (((b4 & 0xff) << 4) << 1);
  }
  
  return (result&0xffff).toString(16).padStart(4, '0');
}

const dump = (data) => {
    str = 'fe ';
    str += crc16(data);
    str += ' ';
    str += toHexString([data.length]);
    str += ' ';
    for (const b of data) {
        str += toHexString([b]) + ' ';
    }
    // console.log(str);
}

idx = 0;
bb = [];
for (const b of binary) {
    bb.push(b);
    if (idx++ == 255) {
        // dump(bb)
        // idx = 0;
        // bb = [];
    }    
}
// dump(bb)

console.log(JSON.stringify(bb))

File: crc.py
def _hex(h,n):
    return "{0:0{1}x}".format(h,n)

def crc(bytes):
    result = 65258
    for b in bytes:
        b2 = (((result & 0xff) << 8) | ((0xFF00 & result) >> 8)) ^ (b & 0xff)
        b3 = b2 ^ ((b2 & 0xff) >> 4)
        b4 = b3 ^ (((b3 & 0xff) << 8) << 4)
        result = b4 ^ (((b4 & 0xff) << 4) << 1)

    return _hex(result & 0xffff, 4)

def dump(bytes, l):
    str = 'fe ';
    str = str + crc(bytes);
    str += ' ';
    str += _hex(l & 0xff, 2);
    for c in bytes:
        str += ' ' + _hex(c, 2)
    return str

def read_file(f):
    lines = []
    cleaned = []
    bytes = []
    l = 0
    size = 0
    idx = 0;
    with open("./source/432.bin", "rb") as f:
        l = f.read(None)
        for c in l:
            bytes.append(c)
            idx += 1
            if idx > 255:
                lines.append(dump(bytes, idx))
                bytes = []
                size += idx
                idx = 0
        if (idx > 0):
            lines.append(dump(bytes, idx))
            bytes = []
            size += idx
            idx = 0

    res = ""
    for l in lines:
        res += l + " \n"
    print(res)
    # print(_hex(size,4))
    return [ _hex(size,6), lines ]
    
# read_file('./source/432.bin')
read_file('./source/1472.bin')